io.openems.edge.meter.schneider.acti9.smartlink/src/io/openems/edge/meter/schneider/acti9/smartlink/MeterSchneiderActi9Smartlink.java

```diff
@@ -99,20 +100,27 @@ protected ModbusProtocol defineModbusProtocol() throws OpenemsException {
 						m(AsymmetricMeter.ChannelId.VOLTAGE_L3, new FloatDoublewordElement(3032 - offset),
 								ElementToChannelConverter.SCALE_FACTOR_3)),
 				new FC4ReadInputRegistersTask(3054 - offset, Priority.HIGH,
-						m(AsymmetricMeter.ChannelId.ACTIVE_POWER_L1, new FloatDoublewordElement(3054 - offset)),
-						m(AsymmetricMeter.ChannelId.ACTIVE_POWER_L2, new FloatDoublewordElement(3056 - offset)),
-						m(AsymmetricMeter.ChannelId.ACTIVE_POWER_L3, new FloatDoublewordElement(3058 - offset)),
-						m(SymmetricMeter.ChannelId.ACTIVE_POWER, new FloatDoublewordElement(3060 - offset)),
+						m(AsymmetricMeter.ChannelId.ACTIVE_POWER_L1, new FloatDoublewordElement(3054 - offset),
+								ElementToChannelConverter.INVERT_IF_TRUE(inverted)),
+						m(AsymmetricMeter.ChannelId.ACTIVE_POWER_L2, new FloatDoublewordElement(3056 - offset),
+								ElementToChannelConverter.INVERT_IF_TRUE(inverted)),
+						m(AsymmetricMeter.ChannelId.ACTIVE_POWER_L3, new FloatDoublewordElement(3058 - offset),
+								ElementToChannelConverter.INVERT_IF_TRUE(inverted)),
+						m(SymmetricMeter.ChannelId.ACTIVE_POWER, new FloatDoublewordElement(3060 - offset),
+								ElementToChannelConverter.INVERT_IF_TRUE(inverted)),
 						new DummyRegisterElement(3062 - offset, 3067 - offset),
-						m(SymmetricMeter.ChannelId.REACTIVE_POWER, new FloatDoublewordElement(3068 - offset))),
+						m(SymmetricMeter.ChannelId.REACTIVE_POWER, new FloatDoublewordElement(3068 - offset),
+								ElementToChannelConverter.INVERT_IF_TRUE(inverted))),
 				new FC4ReadInputRegistersTask(3110 - offset, Priority.LOW,
 						m(SymmetricMeter.ChannelId.FREQUENCY, new FloatDoublewordElement(3110 - offset),
 								ElementToChannelConverter.SCALE_FACTOR_3)),
 				new FC4ReadInputRegistersTask(3208 - offset, Priority.LOW,
```

Comment:
For the energy values, simply inverting will not be sufficient. The energy channel values are expected to be steadily increasing.

See this example: https://github.com/OpenEMS/openems/blob/develop/io.openems.edge.meter.socomec/src/io/openems/edge/meter/socomec/threephase/SocomecMeterThreephaseImpl.java#L136 and this discussion: https://community.openems.io/t/schneider-zahlerdaten-invertieren/322/2

Reply:
Thanks